src/url.js

```diff
@@ -0,0 +1,27 @@
+import isUrl from 'is-url';
+
+import wrapValidator from './helpers/wrapValidator';
+
+function url(props, propName, componentName) {
+  const value = props[propName];
+
+  if (value == null || isUrl(value)) {
+    return null;
+  }
+
+  return new TypeError(`${propName} in ${componentName} must be a valid URL; value is ${value}`);
```

Comment:
Not crazy about the format of either of these error messages. Happy to discuss alternative formats though I think it's useful to include the invalid value in the error message.